GMG/GMG/UI/Components/Metronome.swift

```diff
@@ -6,13 +6,164 @@
+            }
+            audioPlayer.currentTime = 0
+            audioPlayer.play()
+        } else {
+            guard let audioPlayer = lowPlayer else {
+                print("[Metronome] lowPlayer is nil")
+                return
+            }
+            audioPlayer.currentTime = 0
+            audioPlayer.play()
+        }
+    }
+
     var body: some View {
-        Text(/*@START_MENU_TOKEN@*/"Hello, World!"/*@END_MENU_TOKEN@*/)
+        
+        //TimelineView를 사용해 beatInterval마다 뷰가 업데이트 되도록 구성함
+        TimelineView(.periodic(from: .now, by: beatInterval)) { context in
+            BeatBar(beatsPerBar: beatsPerBar, currentBeat: currentBeatToShow)
+                .animation(.spring(response: 0.28, dampingFraction: 0.85, blendDuration: 0.2), value: beatsPerBar)
+                .padding()
+                .onChange(of: context.date) { _, _ in
+                    guard isPlaying, let start = startDate else { return }
+                    let step = Int(floor(Date().timeIntervalSince(start) / beatInterval))
+                    if step != lastStepPlayed {
+                        lastStepPlayed = step
+                        playBeatSound(for: currentBeatToShow)
+                    }
+                }
+        }
+        .onAppear {
+            configureAudioSession()
+            loadPlayers()
+        }
+        .onDisappear() {
+            // audio session 관련 메모리 해제 로직
+            // 1. 플레이어 중지 및 메모리에서 해제
+            highPlayer?.stop()
+            lowPlayer?.stop()
+            highPlayer = nil
+            lowPlayer = nil
+            
+            // 2. 오디오 세션 비활성화
+            do {
+                try AVAudioSession.sharedInstance().setActive(false, options: .notifyOthersOnDeactivation)
+            } catch {
+                print("[Metronome] AudioSession deactivation error: \(error)")
+            }
+        }
+        //박자가 변경될 때 자연스럽게 CD 갯수가 바뀌는 애니메이션
+        .onChange(of: selectedTimeSig) { _, _ in
+            withAnimation(.spring(response: 0.28, dampingFraction: 0.85, blendDuration: 0.2)) {
+                if isPlaying { startDate = Date(); lastStepPlayed = -1 }
+            }
+        }
+        .onChange(of: isPlaying) { _, newValue in
+            if newValue {
+                startDate = Date()
+                lastStepPlayed = -1
+            }
+        }
+        .onChange(of: bpm.value) { _, _ in
+            if isPlaying { startDate = Date(); lastStepPlayed = -1 }
+        }
+    }
+}
+
+extension Metronome {
+    struct BeatBar: View {
+        let beatsPerBar: Int
+        let currentBeat: Int
+        
+        var body: some View {
+            HStack(spacing: beatsPerBar == 6 ? 18 : 24) {
+                ForEach(0 ..< beatsPerBar, id: \.self) { index in
+                    Image(index == currentBeat ? "metronome_cd_fill" : "metronome_cd_empty")
+                        .resizable()
+                        .scaledToFit()
+                        .frame(height: (beatsPerBar == 6 ? 33 : 44))
```

Comment:
spacing이랑 height는 변수로 빼는 게 좋을 것 같습니다.

```suggestion
        private var spacing: CGFloat { beatsPerBar == 6 ? 18 : 24 }
        private var height: CGFloat { beatsPerBar == 6 ? 33 : 44 }
        
        var body: some View {
            HStack(spacing: spacing) {
                ForEach(0 ..< beatsPerBar, id: \.self) { index in
                    Image(index == currentBeat ? "metronome_cd_fill" : "metronome_cd_empty")
                        .resizable()
                        .scaledToFit()
                        .frame(height: height)
```

Reply:
바뀐 디자인에도 수정해서 적용했습니다!